services/cd-service/pkg/repository/repository.go

```diff
@@ -1135,7 +1133,12 @@ func extractPrNumber(sourceMessage string) string {
 }
 
 func (s *State) IsUndeployVersion(ctx context.Context, transaction *sql.Tx, application string, version uint64) (bool, error) {
-	release, err := s.DBHandler.DBSelectReleaseByVersion(ctx, transaction, application, version, true)
+	release, err := s.DBHandler.DBSelectReleaseByVersion(ctx,
```

Comment:
I'd prefer if this function also gets the revision number, and if it's not 0, you can immediately return without doing the query.